experimental/dds/tree2/src/core/tree/delta.ts

```diff
@@ -167,226 +95,95 @@ export type ProtoNode = ITreeCursorSynchronous;
-	 * ID associated with the tree being built.
-	 * If not specified, a random unique ID will be used.
-	 */
-	readonly buildId?: DetachedNodeId;
-
-	/**
-	 * The ID to assign the first node being replaced.
-	 * Subsequent replaced nodes should be assigned incrementing IDs.
-	 * Populated iff the insertion is transient.
-	 */
-	readonly detachId?: DetachedNodeId;
+export interface DetachedNodeId {
+	readonly major?: string | number;
+	readonly minor: number;
 }
 
 /**
- * Describes the restoration of a contiguous range of node.
  * @alpha
  */
-export interface Restore<TTree = ProtoNode> extends CanReplaceContent<TTree> {
-	readonly type: typeof MarkType.Restore;
-	/**
-	 * Must be 1 when `fields` is populated.
-	 */
-	readonly count: number;
-
-	readonly newContent: {
-		/**
-		 * The ID assigned to the first node being restored.
-		 * Subsequent nodes should be assigned incrementing IDs.
-		 */
-		readonly restoreId: DetachedNodeId;
-
-		/**
-		 * The ID to assign the first node being replaced.
-		 * Subsequent replaced nodes should be assigned incrementing IDs.
-		 * Populated iff the restoration is transient.
-		 */
-		readonly detachId?: DetachedNodeId;
-	};
-
-	/**
-	 * Modifications to the new content.
-	 */
-	readonly fields?: FieldMarks<TTree>;
-}
+export type FieldMap<T> = ReadonlyMap<FieldKey, T>;
 
 /**
- * Uniquely identifies a node in the scope of a delta.
- * Often accompanied with a count that indicates a set of nodes whose IDs are contiguous.
  * @alpha
  */
-export interface MoveId extends Opaque<Brand<number, "delta.MoveId">> {}
+export type FieldsChanges<TTree = ProtoNode> = FieldMap<FieldChanges<TTree>>;
 
 /**
- * A globally unique ID for a node in a detached field.
+ * Represents changes made to a detached node
  * @alpha
  */
-export interface DetachedNodeId {
-	major?: string | number;
-	minor: number;
+export interface DetachedNodeChanges<TTree = ProtoNode> {
+	readonly id: DetachedNodeId;
+	readonly fields: FieldsChanges<TTree>;
 }
 
 /**
+ * Represents the creation of a detached node
  * @alpha
  */
-export type FieldMap<T> = ReadonlyMap<FieldKey, T>;
+export interface DetachedNodeBuild<TTree = ProtoNode> {
+	readonly id: DetachedNodeId;
+	readonly trees: readonly TTree[];
+}
 
 /**
+ * Represents a detached node being assigned a new `DetachedNodeId`.
  * @alpha
  */
-export type FieldMarks<TTree = ProtoNode> = FieldMap<MarkList<TTree>>;
+export interface DetachedNodeRename {
+	readonly count: number;
+	readonly oldId: DetachedNodeId;
+	readonly newId: DetachedNodeId;
+}
 
 /**
+ * Represents the changes to perform on a given field.
  * @alpha
  */
-export const MarkType = {
-	Modify: 0,
-	Insert: 1,
-	MoveIn: 2,
-	Remove: 3,
-	MoveOut: 4,
-	Restore: 5,
-} as const;
+export interface FieldChanges<TTree = ProtoNode> {
+	readonly attached?: MarkList<TTree>;
+	readonly detached?: readonly DetachedNodeChanges<TTree>[];
+	readonly build?: readonly DetachedNodeBuild<TTree>[];
+	readonly rename?: readonly DetachedNodeRename[];
```

Comment:
I think we should make the grammatical case consistent for these field names. How about build -> built and rename -> renamed?

Reply:
I guess I see why they are in different cases, but maybe we should make the names more verbose. Maybe something like `attachedChanges`, `detachedChanges`, `builds`, and `renames`?

Reply:
I was worried this would be confusing. I find `attachedChanges` a bit confusing as well since it's not the changes that are attached/detached (`attachedNodesChanges` would be clearer but it's getting rather long which will make the code formatting ugly). How about `inlineChanges` and `rootChanges`?